File: src/Cadastro.API/Models/Request/EnderecoRequest.cs
namespace Cadastro.API.Models.Request
{
    public class EnderecoRequest
    {
        public int Id { get; set; }
        public string Rua { get; set; }
        public int? Numero { get; set; }
        public string CEP { get; set; }
        public string Complemento { get; set; }
        public string Bairro { get; set; }
        public string Cidade { get; set; }
        public string UF { get; set; }
    }
}

File: src/cadastro.angular/src/app/models/request/enderecoRequest.ts
import { EnderecoResponse } from "../response/enderecoResponse";

export class EnderecoRequest {
    id!: number;
    rua!: string;
    numero!: number | null;
    cep!: string;
    complemento!: string;
    bairro!: string;
    cidade!: string;
    uf!: string;

    constructor(endereco: EnderecoResponse) {
        this.id = endereco.id;
        this.rua = endereco.rua;
        this.numero = endereco.numero;
        this.cep = endereco.cep;
        this.complemento = endereco.complemento;
        this.bairro = endereco.bairro;
        this.cidade = endereco.cidade;
        this.uf = endereco.uf;
    }
}